src/components/Modal/ExternalLinkWarningModal.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useCallback, useEffect, useState } from 'react'
+
+import Paragraph from 'decentraland-gatsby/dist/components/Text/Paragraph'
+import useFormatMessage from 'decentraland-gatsby/dist/hooks/useFormatMessage'
+import { Button } from 'decentraland-ui/dist/components/Button/Button'
+import { Close } from 'decentraland-ui/dist/components/Close/Close'
+import { Header } from 'decentraland-ui/dist/components/Header/Header'
+import { Modal, ModalProps } from 'decentraland-ui/dist/components/Modal/Modal'
+
+import './ExternalLinkWarningModal.css'
+
+type Props = Omit<ModalProps, 'children'>
+
+type WarningModalProps = {
+  isWarningModalOpen: boolean
+  href: string
+}
+
+function openInNewTab(url: string) {
+  window?.open(url, '_blank')?.focus()
+}
+
+const WHITELIST = [
+  /^https:\/\/([a-zA-Z0-9]+\.)?decentraland\.org(\/[^\s]*)?$/g,
+  /^https:\/\/([a-zA-Z0-9]+\.)?decentraland\.vote(\/[^\s]*)?$/g,
+  /^https:\/\/([a-zA-Z0-9]+\.)?snapshot\.org(\/[^\s]*)?$/g,
+  /^https:\/\/([a-zA-Z0-9]+\.)?dcl\.gg(\/[^\s]*)?$/g,
+]
+
+const INITIAL_STATE: WarningModalProps = {
+  isWarningModalOpen: false,
+  href: '',
+}
+
+function ExternalLinkWarningModal({ ...props }: Props) {
+  const t = useFormatMessage()
+  const [warningModalProps, setWarningModalProps] = useState(INITIAL_STATE)
```

Comment:
Now that they aren't props maybe this name could be `warningModalState`

Reply:
Makes sense, I'll change it